Resources/doc/cookbook/recipe_hashid.rst

```diff
@@ -0,0 +1,118 @@
+also an ``encode`` function. For more information on this see the
+documentation in the `HashId Symfony Bundle`_
+
+.. code:: php
+
+    class MyAdmin extends AbstractAdmin
+    {
+        /**
+         * @var HashId
+         */
+        private $hashId
+
+        /**
+         * @param string $code
+         * @param string $class
+         * @param string $baseControllerName
+         * @param HashId $hashId
+         */
+        public function __construct($code, $class, $baseControllerName, HashId $hashId)
+        {
+            parent::__construct($code, $class, $baseControllerName);
+            $this->hashId = $hashId
+        }
+
+        ... other configurations
+
+        /**
+         * @param string $name
+         * @param mixed $object
+         * @param array $parameters
+         * @param bool|int $absolute
+         * @return string
+         */
+        public function generateObjectUrl($name, $object, array $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH)
+        {
+            $parameters['id'] = $this->hashId->encodeHex($this->getUrlsafeIdentifier($object));
+
+            return $this->generateUrl($name, $parameters, $absolute);
+        }
+    }
+
+
+Now your routes will change from ``/product/1/edit`` to
+``/product/af5/edit``, child routes will also be changed if you have
+chosen to use them e.g. ``product/af5/child/e6D/edit``. So now you will
+have to tell your controllers to decode these. For this implement a
+class that uses the ``RouteIdHandlerInterface`` to decode the id held
+in the request. For example
+
+.. code:: php
+
+    class HashIdRouteIdHandler implements RouteIdHandlerInterface
+    {
+        /**
+         * @var HashId
+         */
+        private $hashId
+
+        /**
+         * @param HashId $hashId
+         */
+        public function __construct(HashId $hashId)
+        {
+            $this->hashId = $hashId
+        }
+
+        /**
+         * @param Request $request
+         * @param AdminInterface $admin
+         *
+         * @return int|string
+         */
+        public function getIdFromRequest(Request $request, AdminInterface $admin)
+        {
+            return $this->hashId->decodeHex($request->get($admin->getIdParameter()));
+        }
+    }
+
+Add this into your relevant services file and then you will can either
+override this on a per Controller basis where you need it. e.g.
+
+.. code:: php
+
+    class MyCRUDController extends CRUDController
+    {
+        /**
+         * @return int|string
+         */
+        protected function getObjectIdFromRouteIdHandler()
+        {
+            return $this->get('hashid_route_id_handler')
+                ->getIdFromRequest($this->getRequest(), $this->admin);
+        }
+    }
+
+Or if you you want this to always use hashes then then you can change this in the config
+
+.. code:: php
+
+    # config.yml
+
+    sonata_admin:
+        route_id_handler: hashid_route_id_handler
```

Comment:
Oh I didn't see that part. So I guess this means the configuration already works, great!

Reply:
We try to discourage inheritance since recently, so please remove the paragraph about overriding, and suggest determining if decoding is needed right in the service, based on the admin object (you could check for an interface it would implement, for instance)

Reply:
Do you mean in the service you would do something like this?
```
if($admin instanceof HashedAdminInterface) {
    return $this->hashId->decodeHex($request->get($admin->getIdParameter()));
} else {
    return $request->get($admin->getIdParameter()));
}
```
If so I might add that as a suggestion.

Reply:
Yes, that's what I mean (but remove the `else`, there is a `return`, so no need for it). And please make it clear that any condition can be used, and that `HashedAdminInterface` does not exist in this repo.

Reply:
Thanks for the clarification, I'll get that in

Reply:
Oh I'm going to keep the `else` in because this is supposed to be a selective example. As in not all Admins want to use the HashId so it needs a fallback for any Admin that does not implement that interface

Reply:
I mean this:
```php
<?php
if ($admin instanceof HashedAdminInterface) {
    return $this->hashId->decodeHex($request->get($admin->getIdParameter()));
}

return $request->get($admin->getIdParameter()));
```

Reply:
Fair enough, consider it done